### ISSUE TITLE
Infrataster doesn't work with ports other than 80

### ISSUE THREAD

**Issue by estahn** [zuazo/dockerspec#5]
### Dockerspec Version
0.4.1

### Ruby Version
ruby 2.4.0p0

### Platform Details
Mac

### Scenario
Testing a docker container exposing port 9000 with Infrataster.

### Steps to Reproduce
```ruby
    describe server(described_container) do

      describe http('/healthcheck') do
        it 'responds with works' do
          expect(response.body).to include 'WORKS'
        end
      end

    end
```

### Expected Result
```
Serverspec on tag: "sha256:21afedc21ede0f4fb2e6416d83cf2f70b60e3fa3a6c275e1ec53a8f4f78e4e81"
    server '/sleepy_bose'
      http 'http://172.0.1.2:9000/healthcheck' with {:params=>{}, :method=>:get, :headers=>{}}
```

### Actual Result
```
Serverspec on tag: "sha256:21afedc21ede0f4fb2e6416d83cf2f70b60e3fa3a6c275e1ec53a8f4f78e4e81"
    server '/sleepy_bose'
      http 'http:/healthcheck' with {:params=>{}, :method=>:get, :headers=>{}}
```

**Comment by zuazo** [comment]
Hi @estahn,

Try passing the complete URL to the `http` resource:

```ruby
    describe server(described_container) do

      describe http('http://localhost:9000/healthcheck') do
        it 'responds with works' do
          expect(response.body).to include 'WORKS'
        end
      end

    end
```

That should work in theory. Let me know if not.

**Comment by estahn** [comment]
It doesn't because the port is not mapped. dockerspec will need to look for
a random available port and map the exposed container port. Then specify it
in the `server` resource.

Also I'd like to avoid specifying the entire URL otherwise dockerspec
becomes useless in this case.

Note: I currently use Infrataster without dockerspec and it works. I'm
trying to integrate it in the context of dockerspec.

**Comment by zuazo** [comment]
Sorry @estahn, I misunderstood the issue.

Please, could you give me an example of Dockerfile/docker-compose.yml to reproduce the problem?

**Comment by estahn** [comment]
@zuazo No worries. I don't use docker-compose, but i might start with it for my test cases.

The `_spec` files looks like this:
```ruby
require 'dockerspec/infrataster'

describe 'thumbor' do
  before :all do
    image = Docker::Image.get(ENV['DOCKER_IMAGE'])

    set :os, family: :alpine
    set :backend, :docker
    set :docker_image, image.id
  end

  describe docker_run(tag: Docker::Image.get(ENV['DOCKER_IMAGE']).id) do
    describe server(described_container) do
      describe http("/healthcheck") do
        it 'responds with works' do
          expect(response.body).to include 'WORKS'
        end
      end

    end
  end

end
```

Call:
```
DOCKER_IMAGE=hipages/thumbor:6.3 bundle exec rspec
```

The docker image needs to be pulled before, but is currently not publicly available. You can use `zanui/thumbor` instead.

There is another issue which is kinda related to the port problem. It's not possible to access container via IP and port directly on Mac.

The current workaround i'm testing is from my old infrataster tests:
```
set :docker_container_create_options, {
  "AttachStdout" => true,
  "AttachStderr" => true,
  "PortBindings" => { "9000/tcp" => [{ "HostIp" => "0.0.0.0", "HostPort" => "9000" }]}
}
```

Basically exposing the port the host system like you suggested.

Any better idea would be appreciated.

**Comment by estahn** [comment]
I got it working with some ugly workaround.

```
require 'dockerspec/infrataster'

Infrataster::Server.define('app', '127.0.0.1')
set :docker_container_create_options, {
  "PortBindings" => { "9000/tcp" => [{ "HostIp" => "0.0.0.0", "HostPort" => "9000" }]}
}

describe 'thumbor' do
  before :all do
    image = Docker::Image.get(ENV['DOCKER_IMAGE'])

    set :os, family: :alpine
    set :backend, :docker
    set :docker_image, image.id
  end

  describe docker_run(tag: Docker::Image.get(ENV['DOCKER_IMAGE']).id, wait: 1) do
    describe server('app') do
      describe http("http://app:9000/healthcheck") do
        it "responds content including 'WORKING'" do
          expect(response.body).to include('WORKING')
        end
      end
    end
  end
end
```
